src/orion/core/io/experiment_builder.py

```diff
@@ -855,11 +919,14 @@ def _default(v: T | None, default: V) -> T | V:
 
         space = _instantiate_space(space)
         max_trials = _default(max_trials, orion.core.config.experiment.max_trials)
+        if isinstance(knowledge_base, dict):
+            knowledge_base = _instantiate_knowledge_base(knowledge_base)
```

Comment:
Is this part really not covered in tests? The KB is never defined as a dict?

Reply:
Should be fixed by 34965b402c81ab37aad27c83e034780bc2e0f162